Nuxt init

cbasje opened the pull request.

cbasje committed:
Initial commit

cbasje committed:
Initial commit

cbasje committed:
Update .gitignore

cbasje committed:
Added logo

cbasje committed:
Added favicon

cbasje committed:
Added Pinia Store + main layout

cbasje committed:
Updated main layout

cbasje committed:
Started data structure

cbasje committed:
Added vuedraggable

cbasje committed:
Design update

cbasje committed:
Fixed navigation guard

cbasje committed:
Added Sliding panel

cbasje committed:
Added SlideOver

cbasje committed:
Several small clean ups

cbasje committed:
Questions with data structure

RensHam commented on types/block.model.ts:
```diff
@@ -0,0 +1,26 @@
+import { RoundDetail } from './round.model';
+
+export interface Block {
+	id: string; // Zijn alle '_oud' ids nog nodig?
```
De _oud zijn niet meer nodig en worden niet gebruikt was alleen nodig tijdens een migratie.

RensHam commented on types/block.model.ts:
```diff
@@ -0,0 +1,26 @@
+	block: number;
+	start_time: Date;
+	status: number; // Wat zijn de waarden van deze statussen?
+	actions: Action[]; // Kan dit in de frontend?
```
Dit hoort in de frontend

RensHam commented on types/block.model.ts:
```diff
@@ -0,0 +1,26 @@
+	regatta_id: string;
+	block: number;
+	start_time: Date;
+	status: number; // Wat zijn de waarden van deze statussen?
```
Zie: https://github.com/Proteus-Eretes/IRIS2/blob/develop/src/Model/Types/Block/BlockStatus.php

Als je het niet kan zien:

```
 public const CREATED = 1;
    public const ROUNDS_ASSIGNED = 2;
    public const EVENTS_ASSIGNED = 4;
    public const START_ORDER = 8;
    public const TOSSED = 16;
    public const SHIRT_NUMBERS_ASSIGNED = 32;

    public const BLOCK_STATUS = [
        self::CREATED => 'Created',
        self::ROUNDS_ASSIGNED => 'Rounds assigned',
        self::EVENTS_ASSIGNED => 'Events assigned',
        self::START_ORDER => 'Starting order determined',
        self::TOSSED => 'Lots drawn',
        self::SHIRT_NUMBERS_ASSIGNED => 'Shirt numbers assigned',
    ];
```

cbasje replied:
Ah ja, bdeankt! Ik zal sws de types in die map even overnemen

RensHam commented on types/club.model.ts:
```diff
@@ -0,0 +1,24 @@
+	regatta_id: string;
+	knrb_num: number;
+	code: string;
+	name: string; // Is dit hetzelfde als 'code'?
```
Nee als het goed is niet

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	club_id: string;
+	regatta_id: string;
+	event_id: string;
+	sub_event_id: null; // Wat is een sub_event?
```
Dit is een string|null:

Een ploeg kan een sub_event_id hebben dit treed op als er te weinig ploegen zijn voor een eigen veld. Dus ze starten dan door elkaar maar willen wel een eigen uitslag hebben. (Er zitten wat meer regels aan)

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	sub_event_id: null; // Wat is een sub_event?
+	subEventCode: string;
+	knrb_num: number;
+	knrb_sequence: number; // Wat is de sequence?
```
Bij het inschrijven bij de wedstrijd krijg je bijvoorbeeld:

Proteus 1
Proteus 2
ect

De sequence is dit nummer.

cbasje replied:
Maar er is ook een 'sequence' hier? Is dat dan hetzelfde?

RensHam replied:
knrb_sequence komt uit de import. Deze veranderd niet maar de er kunnen ploegen bij komen of velden samenvoegen

cbasje replied:
Kan dat dan niet worden samengevoegd of 'berekend' door cake?

RensHam replied:
Samengevoegd kan niet en berekend kan ook niet omdat je de volgorde van de ploegen overhoopt gooit bij de loting

RensHam replied:
Sequence mag dan ook niet worden aangepast door de wl en hoeft niet getoond te worden

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	status: number;
+	result_status: null;
+	clubName: string;
+	displayName: string; // Wat is het verschil tussen 'displayName' en 'name'?
```
displayName is de naam die weer geven word. Dit is afhankelijk van wedstrijd instellingen maar ongeveer:

if (alternative) displayName = alternative
elseif (sequence) displayName = club_name + sequence
else displayName = club_name

Wat in de name staat is volgens mij de clubname weet ik niet zeker.

RensHam replied:
Graag alleen de displayName gebruiken

cbasje replied:
Is de 'clubName' dan nog nodig als het hetzelfde is als 'name'?

RensHam replied:
clubname word op de PDF getoond voor de head of the river bijvoorbeeld. Bij de PEiL is het de display name

cbasje replied:
Ja oké, maar er is nu 'name', 'clubName' en 'displayName'

RensHam replied:
name is niet relevant.

clubName is voor de wedstrijden als head of the river
displayName is voor studenten compementie wedstrijden

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	clubName: string;
+	displayName: string; // Wat is het verschil tussen 'displayName' en 'name'?
+	name: string;
+	shortname: string; // Wat is het verschil tussen 'shortname' en 'alternative'?
```
shortname is gemaakt door de KNRB, alternative is ingevuld door de ploeg

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	alternative: string;
+	sequence: number;
+	combination: null;
+	boat: string; // Wat is boat en boattype?
```
Dit is de boot nummer die ze hebben

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	sequence: number;
+	combination: null;
+	boat: string; // Wat is boat en boattype?
+	boattype: string;
```
Dit zou 4+ of iets moeten zijn

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	combination: null;
+	boat: string; // Wat is boat en boattype?
+	boattype: string;
+	remarks: null; // Wat is het verschil tussen 'remarks' en 'public_remarks'?
```
Remarks voor kamprechters, public remarks komen op hoesnelwasik.nl

cbasje replied:
Als 'remarks' voor kamprechters zijn, wat is 'remarks_for_umpire' dan?

RensHam replied:
Oja remarks zijn voor de wedstrijdleiding,
remarks_for_umpire -> kamprechters

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	penalty_reason: null;
+	correction_factor: number;
+	category: string;
+	calculated_category: string; // Wat is het verschil tussen 'category' en 'calculated_category'?
```
Category is waar de ploeg in wil starten.
calculated_category is waar de ploeg in thuis hoort.


Bijvoorbeeld

Stel je bent als ploeg 60 jaar oud dus category -> E
Maar je wil toch liever in D starten dan mag dat.
Maar je mag dan niet stiekem naar category F toe.

Dit word gebruik bij tweede import intern maar ook voor het tonen van errors/warnings

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	correction_factor: number;
+	category: string;
+	calculated_category: string; // Wat is het verschil tussen 'category' en 'calculated_category'?
+	local: number; // Wat is dit?
```
Als een ploeg handmatig is toegevoegd of niet of lokaal teveel gewijzd

cbasje replied:
Is dit een enum die ik ergens kan vinden?

RensHam replied:
Het is een boolean 0/1

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	calculated_category: string; // Wat is het verschil tussen 'category' en 'calculated_category'?
+	local: number; // Wat is dit?
+	payments?: any[];
+	crews_data?: any[]; // Wat is dit?
```
Hier in kan random data staan:

Key value pair. word niet echt gebruikt

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	rowers?: Rower[];
+	club?: Club;
+	event?: Event;
+	sub_event?: null; // Wat is het verschil tussen 'sub_evetn' en 'event'?
```
Zie sub_event_id voor uitleg

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	payments?: any[];
+	crews_data?: any[]; // Wat is dit?
+	fines?: any[];
+	coxes?: any[]; // Kan dit niet samen met 'rowers' omdat het enige verschil de rol is?
```
Daar moet ik naar kijken.

Dan zou het members worden en staan de coaches er ook in

RensHam replied:
De `members` veld bestaat als dat wenselijk kan dat toegevoegd worden.
Eventeeul kan de roeiers en stuur ook samen in lijst komen

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	crew: Crew;
+	toss_reason: string | null;
+	shirt_number: number | null;
+	penalty_time: number; // Is dit in seconden?
```
Ja

cbasje replied:
Dus hier niet in milliseconden?

RensHam replied:
Nee helaas niet

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	penalty_time: number; // Is dit in seconden?
+	penalty_reason: string;
+	deleted: boolean;
+	user_set_deleted: boolean; // Wat is het verschil met deleted?
```
Dit heeft met interne logica te maken omtrent tweede imports. Laat maar weten of je detail uitleg nodig hebt

cbasje replied:
Oké, voor nu is het misschien niet nodig dan haha

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,64 @@
+	deleted: boolean;
+	user_set_deleted: boolean; // Wat is het verschil met deleted?
+	equal_time_correction: string;
+	results: Result[]; // Hoe ziet een result er uit?
```
```
 * @property string $team_id
 * @property string $time
 * @property int $deviation
 * @property int $pulses
 * @property int $status
 * @property string $location_id
 * @property string $round_id
 ```

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	id: string;
+	regatta_id: string;
+	cost_id: string;
+	parent: null; // Wat is de 'parent'?
```
Dat is parent event nodig voor sub_events

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	cost_id: string;
+	parent: null; // Wat is de 'parent'?
+	event: number;
+	number: number; // Is dit hetzelfde als 'event'?
```
Nummer is ingesteld door wedstrijd leiding. Andere is door de knrb

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	parent: null; // Wat is de 'parent'?
+	event: number;
+	number: number; // Is dit hetzelfde als 'event'?
+	day: Date; // Wordt een event op alleen een dag gedaan? Is dat anders dan de regatta?
```
Officieel mag een event maar op 1 dag in 1 block gedaan worden.

Dus als de regatta meer daags is dan is een event 1 daags. Dit geld niet voor verschillende afstanden in een meerkamp.

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	number: number; // Is dit hetzelfde als 'event'?
+	day: Date; // Wordt een event op alleen een dag gedaan? Is dat anders dan de regatta?
+	code: string; 
+	name: string; // Is dit hetzelfde als 'name'?
```
Mogelijk klopt de comment hier niet

cbasje replied:
Ah ja, ik bedoelde 'code'

cbasje replied:
Maar ik neem aan dat het dan hetzelfde geval is, dus dat het niet hetzelfde hoeft te zijn

RensHam replied:
Dat klopt ja

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	code: string; 
+	name: string; // Is dit hetzelfde als 'name'?
+	remarks: string;
+	status: DisplayStatus; // Wat zijn de verschillende statussen?
```
public const ONGOING = 0;
    public const CANCELLED = 1;
    public const MERGED = 2;

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	remarks: string;
+	status: DisplayStatus; // Wat zijn de verschillende statussen?
+	category: string;
+	block_knrb: null; // Wat is dit?
```
Dit is nummer|null
Eigenlijk nooit ingesteld maar soms word een event al ingedeeld in blok 2 bijvoorbeeld

cbasje replied:
Kan het dan ook een id zijn ipv een nummer van een block?

RensHam replied:
Nee dit is helaas geen id alleen een nummer zonder verdere info

block 1, block 2 ect.

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	status: DisplayStatus; // Wat zijn de verschillende statussen?
+	category: string;
+	block_knrb: null; // Wat is dit?
+	boat_type: string; // Is dit een enum?
```
Helaas niet

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	weighed: boolean;
+	sex: string;
+	correction_factor_boat: number;
+	sub_crews?: Crew[]; // Wat is het verschil tussen 'sub_crews' en 'crews'?
```
Dit moet ik nakijken

RensHam replied:
Stel je hebt:

V 1x en (VA 1x, VB 1x)

Dan is de crews van VA 1x leeg maar de sub_crews niet

RensHam replied:
Dit is als ik het goed heb

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	round_id: string;
+	starting_order: number | null;
+	result_status: number; // Ook hier wat is de enum met statussen?
+	start_time: null; // Kan dit niet ook op verschillende dagen zijn?
```
Nee dat is niet mogelijk.

RensHam commented on types/event.model.ts:
```diff
@@ -0,0 +1,40 @@
+	regatta_id: string;
+	round_id: string;
+	starting_order: number | null;
+	result_status: number; // Ook hier wat is de enum met statussen?
```
public const DEFAULT = 0;
    public const APPROVED = 1;
    public const UNDER_INVESTIGATION = 2;
    public const FINISHED = 3;
    public const STARTED = 4;
    public const FINISHING = 5;
    public const CANCELED = 6;

RensHam commented on types/regatta.model.ts:
```diff
@@ -0,0 +1,26 @@
+	race_type: string;
+	breaking_news: string;
+	correction_factor_settings: number; // Is dit een object?
+	correction_factor_type: null; // Wat is dit?
```
Dit is een enum

cbasje replied:
Waar kan ik die vinden?

RensHam replied:
Ik zeg het verkeerd de type is uuid en staat in een table.
De settings is bool indien het hard waait of niet

RensHam commented on types/regatta.model.ts:
```diff
@@ -0,0 +1,26 @@
+	end_date: Date;
+	race_type: string;
+	breaking_news: string;
+	correction_factor_settings: number; // Is dit een object?
```
Ik ga dit voor je nakijken

RensHam replied:
Nu niet is niet ingesteld

RensHam commented on types/regatta.model.ts:
```diff
@@ -0,0 +1,26 @@
+	correction_factor_settings: number; // Is dit een object?
+	correction_factor_type: null; // Wat is dit?
+	// Waarom is dit een JSON object als string?
+	lottery_settings: LotterySettings;
```
Omdat dit nooit goed gedaan is :(

RensHam commented on types/result.model.ts:
```diff
@@ -0,0 +1,39 @@
+	pulses: null; // Is dit niet hetzelfde als 'times'?
+	status: number;
+	location: Location;
+	time: string; // Waarom worden hier ineens UNIX gegeven?
```
We hebben twee type pulsen:

start pulse -> unix start tijd van de ploeg
anders pulsen zijn delta tijden

RensHam commented on types/result.model.ts:
```diff
@@ -0,0 +1,39 @@
+	round_id: string;
+	team_id: string;
+	location_id: string;
+	deviation: number; // Is dit seconden?
```
millisecondes

RensHam commented on types/result.model.ts:
```diff
@@ -0,0 +1,39 @@
+	team_id: string;
+	location_id: string;
+	deviation: number; // Is dit seconden?
+	pulses: null; // Is dit niet hetzelfde als 'times'?
```
Ik ga dit nakijken

RensHam replied:
pulses word niet gebruikt volgens mij

RensHam commented on types/result.model.ts:
```diff
@@ -0,0 +1,39 @@
+	assigned: boolean;
+	deleted: boolean;
+	status: string;
+	stopwatch: string; // Wat is dit?
```
Om te weten of deze handmatig is ingevoerd. Dan moet de berekening op een andere manier.

RensHam commented on types/result.model.ts:
```diff
@@ -0,0 +1,39 @@
+
+export interface Time {
+	id: string;
+    // Waarom worden hier ook nog eens de ids gegeven?
```
Intern nodig voor koppeling en performance

RensHam commented on types/round.model.ts:
```diff
@@ -0,0 +1,16 @@
+	id: string;
+	regatta_id: string;
+	name: string;
+	multiplier: number; // Wat is een multiplier?
```
Stel je hebt een afstand van 2500m en 250m dan kan je het terug rekenen naar 500m met de factor: 0.2 en 2

RensHam commented on types/rower.model.ts:
```diff
@@ -0,0 +1,25 @@
+	regatta_id: string;
+	club_id: string;
+	knrb: string;
+	knrb_num: null; // Is dit hetzelfde als 'knrb'?
```
Nee andere nummers

RensHam replied:
knrb_num is het oude systeem en werkt alleen voor Nederlanders

cbasje replied:
Maar knrb_num is nog wel nodig dus?

RensHam replied:
Het is wel nodig aangezien een wedstrijdleiding wel een knrb_num mag invullen maar geen knrb.

RensHam commented on types/rower.model.ts:
```diff
@@ -0,0 +1,25 @@
+	lastname: string;
+	year_of_birth: null;
+	gender: string;
+	_joinData: JoinData; // Wat is JoinData?
```
Dat is cakephp eigen

RensHam replied:
Hier in word informatie van de jointable toegevoegd

RensHam replied:
Dus dat is hier CrewMember of iets anders

cbasje replied:
Is dat handig om te hebben voor de frontend of zal ik het weglaten?

cbasje replied:
Of kan dat niet gewoon worden samengevoegd door cake?

RensHam replied:
Zal ik naar kijken

RensHam commented on pages/blocks/index.vue:
```diff
@@ -0,0 +1,198 @@
+							:key="crew.id"
+						>
+							<!-- <button type="button" @click="selectField(field)"> -->
+							{{ crew.name }}
```
Graag hier de displayName gebruiken

cbasje committed:
Updates nav pull request feedback

cbasje committed:
Update README.md

RensHam commented on types/crew.model.ts:
```diff
@@ -0,0 +1,77 @@
+	shortname: string; // Assigned by KNRB
+	alternative: string; // Created by Crew
+	combination: null;
+	boat: string; // Boat number
```
```suggestion
	boat: string; // Boat id
```

cbasje committed:
Update data structure with server API + Clean up

cbasje committed:
Finish Assign page

cbasje committed:
Added server call to addField

cbasje committed:
Added Toast

Geïnspireerd door, maar aangepast https://github.com/acidjazz/tv-toast/blob/master/src/tv-toast.vue

RensHam reviewed (commented):
Ken je eslint?

cbasje committed:
Implementation of Toast.vue / useToastService

cbasje committed:
Refactoring

cbasje committed:
Refactoring

cbasje committed:
Update index.vue

RensHam commented on components/navigation/SideBarItem.vue:
```diff
@@ -0,0 +1,108 @@
+	item: RouteItem;
+}>();
+
+// FIXME Dit hoort gewoon als string te werken
```
Wat moet hier nog anders?

cbasje committed:
Refactoring

cbasje committed:
Added Table

cbasje committed:
Design update

cbasje committed:
Refactor

cbasje committed:
TODO plugin fix

cbasje committed:
Finished Blocks page

cbasje committed:
Icon update

Icoontjes gebruiken van [phosphoricons.com](https://phosphoricons.com/). Dit heeft meer opties

cbasje committed:
Update index.vue

cbasje committed:
Start to Crews page

cbasje committed:
Added fines

cbasje committed:
Added TablePagination

cbasje committed:
Small fixes

cbasje committed:
Changes after meeting

cbasje committed:
Started Events page

cbasje committed:
Fixes in Events page

cbasje committed:
Added active row highlighting

cbasje committed:
Added option to change spacing in pagination

cbasje committed:
Added collapsable sidebar

cbasje committed:
Bug fixes

cbasje committed:
Update SideBar.vue

cbasje committed:
Update NavigationBar.vue

cbasje committed:
Refactoring code for easy reading + Changes to RowersSlidingPanel

cbasje committed:
Update mock.ts

cbasje committed:
Crews page search

cbasje committed:
Small design changes

cbasje committed:
ESlint

cbasje committed:
Update SearchField.vue

cbasje committed:
Update crews.vue

cbasje committed:
Added sorting to Table

cbasje committed:
Prettier

cbasje committed:
Prettier

cbasje committed:
Crews Add SlideOver

cbasje committed:
Bug fix

cbasje committed:
Rowers Add SlideOver

cbasje committed:
Small design update

cbasje committed:
Update CrewsSlidingPanel

cbasje committed:
BLocks Add SlideOver

cbasje committed:
Fixed sorting

cbasje committed:
Reset on add

cbasje committed:
Events Add SlideOver

cbasje committed:
Fixed Assign

cbasje committed:
Regattas Add/Delete SlideOver

cbasje committed:
Delete actions

cbasje committed:
Regatta Edit SlideOver

cbasje committed:
Fines Delete #5

cbasje committed:
Change crews status #5

cbasje committed:
Update mock.ts

cbasje committed:
Update NavigationBar.vue

cbasje committed:
Prettier

cbasje committed:
Edit actions

cbasje committed:
Refactoring

cbasje committed:
Nuxt middleware #2 + favicon update + Nuxt beta update

cbasje committed:
Documentation

cbasje committed:
Documentatie

cbasje committed:
Small design updates

cbasje committed:
Plugin update

cbasje committed:
Start /lots and /shirt-numbers

cbasje committed:
Small fixes

cbasje committed:
Refactoring because of bugs

cbasje committed:
Fixing Node hydration + refactor files draw

cbasje committed:
Refactoring

cbasje committed:
Fixing toast + adding trycatch

cbasje committed:
Added Draggable Table

cbasje committed:
Added store call

cbasje committed:
Moved overview to separate page

cbasje committed:
Design update

cbasje committed:
Added FormKit

cbasje committed:
Added Accordion + Starting order

cbasje committed:
Added shirt-numbers + starting-order

cbasje committed:
Update starting-order.vue

cbasje committed:
Make sure the regattaId is always loaded

cbasje committed:
Update README.md

cbasje committed:
Update the ShirtNumber panel

cbasje committed:
Added ShirtNumberCalculationSlideOver

cbasje committed:
Deleted comments

cbasje committed:
More explanations

cbasje committed:
Finish draw page
